File: xtask/templates/bridge/apple/Bridge/FlareSdkException.swift
import Foundation

/// Stable Swift error surfaced by the Apple SDK bridge/adapter.
public struct FlareSdkException: Error, LocalizedError, Sendable {
    public let code: String
    public let message: String
    public let operation: String?
    public let details: [String: String]

    public init(code: String, message: String, operation: String? = nil, details: [String: String] = [:]) {
        self.code = code
        self.message = message
        self.operation = operation
        self.details = details
    }

    public var errorDescription: String? {
        if let operation {
            return "FlareSdkException(code=\(code), operation=\(operation), message=\(message))"
        }
        return "FlareSdkException(code=\(code), message=\(message))"
    }
}

File: xtask/templates/bridge/android/bridge/FlareSdkException.kt
package com.flare.im.bridge

/** Stable Android error surfaced by the JNI/FFI bridge/adapter. */
class FlareSdkException(
    val code: String,
    override val message: String,
    val operation: String? = null,
    val details: Map<String, String> = emptyMap(),
) : RuntimeException(message)
